### PR TITLE
Deactivate single product 4219

### PR DESCRIPTION
**Pull request by sandrew**

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
@wstephenson will review your pull request :dancers: check https://trello.com/c/wP3kTFkp/4219-5-sles12sp3-connect-smt-de-activate-one-product-on-a-system
  @wstephenson: Please review this pull request using our guidelines:
  * test for acceptance criteria / functionality
  * check if the new code is covered with tests
  * check for unintended consequences
  * encourage usage of the boyscout rule
  * make sure the code is architected in the best way
  * check that no unnecessary technical debt got introduced
  * make sure that no unnecessary FIXMEs or TODOs got added

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/9290/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 0524130e6c230fdccb1a5c17b8e388f4755d8b1c into a02edebe7825428f8e0a5d20bf3875166f847c87 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/9291/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 0524130e6c230fdccb1a5c17b8e388f4755d8b1c into a02edebe7825428f8e0a5d20bf3875166f847c87 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/9500/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged d956e7f85672c4184f3963a439c5d593de65a387 into a02edebe7825428f8e0a5d20bf3875166f847c87 )

File: lib/suse/connect/templates/extensions_list.text.erb [hunk 1]
```diff
@@ -1,36 +1,23 @@
...
-<% end -%>
-<% end -%>
-
+<%
+any_installed = installed_products.select {|x| !x.isbase }.any?
+{ true => 'FREE EXTENSIONS', false => 'PAID EXTENSIONS' }.each do |type_key, title| %>
+\e[1m<%= title %>\e[0m
+
+<%
+  grouped_extensions[type_key].each do |extension| -%>
```
Comment (ikapelyukhin) [inline lib/suse/connect/templates/extensions_list.text.erb]: That throws an exception when running `SUSEConnect --list-extensions` on a system registered against SMT when `type_key` = `false`.
(It seems there are no paid extensions available specifically for my system)

Here's the trace:
```
SUSEConnect error: NoMethodError: undefined method `each' for nil:NilClass
(erb):9:in `block in render'
(erb):5:in `each'
(erb):5:in `render'
/usr/lib64/ruby/2.1.0/erb.rb:847:in `eval'
/usr/lib64/ruby/2.1.0/erb.rb:847:in `block in result'
/usr/lib64/ruby/2.1.0/erb.rb:848:in `call'
/usr/lib64/ruby/2.1.0/erb.rb:848:in `result'
/root/connect/lib/suse/toolkit/renderer.rb:13:in `render'
/root/connect/lib/suse/connect/status.rb:39:in `print_extensions_list'
/root/connect/lib/suse/connect/cli.rb:31:in `execute!'
./SUSEConnect:11:in `<main>'
```

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-Connect-Prophet/9574/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests.
( Failure: Merged 2c200a20765d3bc23ac7da03ed2fa58d209b84c0 into a02edebe7825428f8e0a5d20bf3875166f847c87 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/10356/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 2c200a20765d3bc23ac7da03ed2fa58d209b84c0 into a02edebe7825428f8e0a5d20bf3875166f847c87 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/13508/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged add0537f265c0d8a816fb5dd2ebfbf24d97babe7 into a4cca0adfa722951303b49559e422b0c885918f8 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/13519/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged d43c2b276ade8db5c1dd1eefba131198b53eb595 into a4cca0adfa722951303b49559e422b0c885918f8 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/13520/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged edb2b84717c89ec57f4748428c3abed19a88a35b into a4cca0adfa722951303b49559e422b0c885918f8 )

File: lib/suse/connect/cli.rb [hunk 2]
```diff
@@ -117,14 +118,17 @@ def extract_options # rubocop:disable MethodLength
...
+        @opts.on('-r', '--regcode [REGCODE]',
+                 'Subscription registration code for the product to',
                  'be registered.',
                  'Relates that product to the specified subscription,',
                  'and enables software repositories for that product.') do |opt|
           @options[:token] = opt
         end
 
-        @opts.on('-d', '--de-register', 'De-registers a system and removes all services',
+        @opts.on('-d', '--de-register',
```
Comment (wstephenson) [inline lib/suse/connect/cli.rb]: --deregister is cleaner here, IMO

File: lib/suse/connect/cli.rb [hunk 3]
```diff
@@ -117,14 +118,17 @@ def extract_options # rubocop:disable MethodLength
...
+                 'De-registers a system or a single product (if',
```
Comment (wstephenson) [inline lib/suse/connect/cli.rb]: 'De-registers the system and base product, or in conjunction with --product, a single addon/extension product, and removes...'

note change to 'the system' not a system in any case.

File: lib/suse/connect/cli.rb (hunk 2 above)
Comment (wstephenson) [inline lib/suse/connect/cli.rb]: My bad, exists already.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-Connect-Prophet/13573/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 2f8d9ac8faf3d88d9d152602e04fc95f5111d8f5 into a4cca0adfa722951303b49559e422b0c885918f8 )

**Pull request merged**